tools/ingest_csv.py

```diff
@@ -0,0 +1,263 @@
+        raise InvalidCSVException(
+            f"Could not find {source_dir}, please make sure to provide a correct source directory."
+        )
+
+    # Extract csv files from source directory
+    csv_files = [f for f in source_dir.iterdir() if f.is_file() and f.suffix == ".csv"]
+    if not csv_files:
+        raise InvalidCSVException(
+            f"No csv files found in {source_dir}, please make sure to provide directory with csv files."
+        )
+
+    # Extract the lineage relationships from the csv files
+    lineages = []
+    for csv_file_to_ingest in csv_files:
+        with open(csv_file_to_ingest, "r") as csv_file:
+            csv_reader = csv.reader(
+                csv_file,
+            )
+            headers = next(csv_reader)
+            index_fullname_src, index_fullname_trg = _validate_header(headers=headers, csv_file=csv_file_to_ingest)
+            unique_asset_types.update(headers[:index_fullname_src])
+            unique_asset_types.update(headers[index_fullname_src + 2 : index_fullname_trg])
+            for line, row in enumerate(csv_reader, start=2):
+                if len(row) != len(headers):
+                    raise InvalidCSVException(
+                        f"""Row {row} (line {line}) in file {csv_file} does not contain same amount
+                         of entries as the header"""
+                    )
+
+                src = _create_asset(
+                    asset_types=headers[:index_fullname_src],
+                    asset_names=row[:index_fullname_src],
+                    fullname=row[index_fullname_src],
+                    domain_id=row[index_fullname_src + 1],
+                    csv_file=csv_file.name,
+                    row=row,
+                    line=line,
+                )
+                trg = _create_asset(
+                    asset_types=headers[index_fullname_src + 2 : index_fullname_trg],
+                    asset_names=row[index_fullname_src + 2 : index_fullname_trg],
+                    fullname=row[index_fullname_trg],
+                    domain_id=row[index_fullname_trg + 1],
+                    csv_file=csv_file.name,
+                    row=row,
+                    line=line,
+                )
+
+                source_code_text, highlights, transformation_display_name = row[index_fullname_trg + 2 :]
+                source_code = _create_source_code(
+                    source_code_text=source_code_text,
+                    highlights=highlights,
+                    transformation_display_name=transformation_display_name,
+                    custom_lineage_config=custom_lineage_config,
+                    line=line,
+                )
+
+                lineages.append(Lineage(src=src, trg=trg, source_code=source_code))
+
+    if custom_lineage_config.dic_info_provided:
+        # collect uuid from DIC
+        asset_types: List[AssetType] = []
+        for asset_type in unique_asset_types:
+            dic_asset_types = collect_assets_typeid(
```

Comment:
do we want to crash if we fail to collect the assets_type_id? Or just issue a warning message? If the second then we would need some try/except.

Also, how many `asset_types` could we expect? Don't we risk spamming the DIC with search requests?

Reply:
If they choose to fetch them from dic I would say we should crash when it fails (having them is mandatory in CL3).

Number of `asset_types`: I don't think that many as this is only for those that customer is creating custom lineage. The other approach would be to fetch them all (and filter in the script) as I don't see a way to provide multiple strings as input to the API (cc @htrungcao )